packages/html-reporter/src/testFileView.tsx

```diff
@@ -87,7 +87,20 @@ function videoBadge(test: TestCaseSummary): JSX.Element | undefined {
 
 function traceBadge(test: TestCaseSummary): JSX.Element | undefined {
   const firstTraces = test.results.map(result => result.attachments.filter(attachment => attachment.name === 'trace')).filter(traces => traces.length > 0)[0];
-  return firstTraces ? <Link href={generateTraceUrl(firstTraces)} title='View trace' className='test-file-badge'>{trace()}</Link> : undefined;
+  if (!firstTraces)
+    return undefined;
+
+  const isFailed = test.outcome === 'unexpected' || test.outcome === 'flaky';
+
+  return <Link
+    href={generateTraceUrl(firstTraces)}
+    title='View trace'
+    className={clsx('test-file-badge', isFailed && 'test-file-trace-button')}>
+    <span className='test-file-trace-button-content'>
+      {trace()}
+      {isFailed && <span className='test-file-trace-button-text'>View Trace</span>}
```

Comment:
I don't think this should be conditional. Either we always show the new button, or we never do.

Reply:
I'm worried that if we always show it, we add too much visual clutter so the highlight doesn't stand out anymore.

Reply:
Maybe "View failing Trace"?

Reply:
Possibly too long, but I guess that's fine. "View Failing Trace" though